lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -335,7 +335,11 @@ protected function fetchVersionValue($versionedClass, $id)
              . ' FROM '  . $tableName
              . ' WHERE ' . implode(' = ? AND ', $identifier) . ' = ?';
 
-        $flatId = $this->identifierFlattener->flattenIdentifier($versionedClass, (array) $id);
+        if (!is_array($id)) {
+            $id = array($this->class->identifier[0] => $id);
```

Comment:
Changing the `assignDefaultVersionValue` signature, simplified a lot! Good advice! :-D